### PR TITLE
Finish lakilester overworld file

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

**Comment by BowserSlug** [comment]
⚠️ This PR introduces 19 warnings:
- src/891b0_len_fb0.c:21: warning: redundant redeclaration of `func_800EFE2C' in same scope
- src/891b0_len_fb0.c:22: warning: redundant redeclaration of `func_800F0248' in same scope
- src/891b0_len_fb0.c:23: warning: redundant redeclaration of `func_800F0490' in same scope
- src/891b0_len_fb0.c:24: warning: redundant redeclaration of `func_800F0864' in same scope
- src/891b0_len_fb0.c:25: warning: redundant redeclaration of `func_800F09EC' in same scope
- src/891b0_len_fb0.c:26: warning: redundant redeclaration of `func_800F0B3C' in same scope
- src/891b0_len_fb0.c:27: warning: redundant redeclaration of `D_80109480' in same scope
- src/891b0_len_fb0.c:28: warning: redundant redeclaration of `D_8010CFF0' in same scope
- src/891b0_len_fb0.c:29: warning: redundant redeclaration of `D_8010CFF4' in same scope
- src/sprite.c:212: warning: comparison between pointer and integer
- src/sprite.c:246: warning: implicit declaration of function `spr_load_sprite'
- src/sprite.c:246: warning: initialization makes pointer from integer without a cast
- src/sprite.c:260: warning: implicit declaration of function `fold_init'
- src/sprite.c:290: warning: initialization from incompatible pointer type
- src/sprite.c:310: warning: implicit declaration of function `spr_update_player_raster_cache'
- src/sprite.c:362: warning: implicit declaration of function `func_802DE780'
- src/sprite.c:389: warning: assignment from incompatible pointer type

File: src/world/partner/lakilester.c [hunk 1]
```diff
@@ -546,8 +543,392 @@ s32 func_802BE6A0_3221F0(f32* arg0) {
...
 
-ApiStatus func_802BE724_322274(Evt* script, s32 isInitialCall);
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BE724_322274);
+ApiStatus func_802BE724_322274(Evt* script, s32 isInitialCall) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    Camera* cameras = &gCameras[CAM_DEFAULT];
+    Npc* npc = script->owner2.npc;
+    s32 colliderHeightTemp;
+    f32 xPosTemp, yPosTemp, zPosTemp, sp2C;
```
Comment (pixel-stuck) [inline src/world/partner/lakilester.c:552]: Maybe better for these to be just x, y, z?

File: src/sprite.c [hunk 2]
```diff
@@ -363,31 +364,38 @@ s32 func_802DE894(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s3
...
+s32 spr_get_npc_raster_info(SpriteRasterInfo* out, s32 npcSpriteID, s32 rasterIndex) {
     SpriteAnimData* sprite;
-    void *temp_v1;
+    unkStructSprite* temp_v1;
+    s32** paletteOffsetCopy;
+    s32 newVar;
+    
 
     sprite = spr_npcSprites[npcSpriteID];
     if (sprite == 0) {
```
[the remark this answers is not in the document]
Comment (Rainchus) [reply src/sprite.c:384]: doing that and removing the pointless if statement does match

File: src/77480.c [hunk 3]
```diff
@@ -1341,7 +1341,7 @@ void render_player_model(void) {
         if (!(playerStatus->flags & PLAYER_STATUS_FLAGS_20000)) {
             if (playerStatus->alpha1 != playerStatus->alpha2) {
                 if (playerStatus->alpha1 < 254) {
-
```
Comment (ethteck) [inline src/77480.c:1344]: These spaces in this file shouldn't be added

File: src/sprite.c [hunk 4]
```diff
@@ -363,31 +355,35 @@ s32 func_802DE894(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s3
 
 INCLUDE_ASM(s32, "sprite", func_802DE8DC);
 
-#ifdef NON_EQUIVALENT
+typedef struct unkStructSprite {
```
Comment (ethteck) [inline src/sprite.c:358]: UnkStructSprite - also I think UnkSpriteStruct is better, cause the other way sounds like this is a sprite and not a struct

File: src/sprite.c [hunk 5]
```diff
@@ -363,31 +355,35 @@ s32 func_802DE894(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s3
...
+    
+    if (sprite != NULL) {
+        paletteOffsetCopy = sprite->palettesOffset;
+        temp_v1 = (unkStructSprite*)sprite->rastersOffset[rasterIndex];
+        out->raster = temp_v1->unk_00;
+        out->width = temp_v1->width;
+        newVar = npcSpriteID;
+        out->height = temp_v1->height;
+        out->defaultPal = paletteOffsetCopy[temp_v1->unk_06];
+        return 1;
```
Comment (ethteck) [inline src/sprite.c:382]: should still return TRUE and FALSE

File: src/world/partner/lakilester.c [hunk 6]
```diff
@@ -18,6 +18,9 @@ BSS f32 D_802BFF28;
 BSS s32 D_802BFF2C;
 BSS unkPartnerStruct D_802BFF30;
 
+void func_800EF3D4(s16);                               /* extern */
```
Comment (ethteck) [inline src/world/partner/lakilester.c:21]: All the m2c comments should be removed

File: src/world/partner/lakilester.c [hunk 7]
```diff
@@ -18,6 +18,9 @@ BSS f32 D_802BFF28;
...
+s32 player_raycast_up_corners(PlayerStatus*, f32*, f32*, f32*, f32*, f32); /* extern */
```
Comment (ethteck) [inline src/world/partner/lakilester.c:22]: Any function declarations from other files like player_raycast_up_corners should go into functions.h and not here in this file

File: src/world/partner/lakilester.c [hunk 8]
```diff
@@ -511,43 +511,424 @@ void func_802BDDD8_321928(Npc* npc) {
...
+    player_raycast_below_cam_relative(&gPlayerStatus, &x, arg0, &z, &colliderHeight, &hitRx, &hitRz,
                                       &hitDirX, &hitDirZ);
 }
 
-ApiStatus func_802BE724_322274(Evt* script, s32 isInitialCall);
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BE724_322274);
+ApiStatus func_802BE724_322274(Evt* script, s32 isInitialCall) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    Camera* cameras = &gCameras[CAM_DEFAULT];
```
Comment (ethteck) [inline src/world/partner/lakilester.c:549]: this is "camera" rather than "cameras"

File: src/world/partner/lakilester.c [hunk 9]
```diff
@@ -511,43 +511,424 @@ void func_802BDDD8_321928(Npc* npc) {
...
+        case 101:
+            sfx_play_sound_at_npc(SOUND_JUMP_2081, 0, NPC_PARTNER);
+            suggest_player_anim_clearUnkFlag(0x10007);
+            /* fallthrough */
+        case 102:
+            D_802BFF14 += 1;
+            /* fallthrough */
+        case 103:
+            if (!(playerStatus->flags & PLAYER_STATUS_ANIM_FLAGS_800)) {
+                npc->pos.x += ((npc->moveToPos.x - npc->pos.x) / npc->duration);
```
Comment (ethteck) [inline src/world/partner/lakilester.c:726]: outer parens on this and the below two lines can be removed

File: src/world/partner/lakilester.c [hunk 10]
```diff
@@ -511,43 +511,424 @@ void func_802BDDD8_321928(Npc* npc) {
...
+                        break;
+                    }
+                }
+            }
+            break;
+        case 3: 
+            npc->flags &= ~NPC_FLAG_40;
+            playerStatus->flags |= NPC_FLAG_100;
+            func_802BD7DC();
+            camYaw = cameras[0].currentYaw;
```
Comment (ethteck) [inline src/world/partner/lakilester.c:820]: all occurrences of `cameras[0]` should instead be `camera->`

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/world/partner/lakilester.c [hunk 11]
```diff
@@ -1,6 +1,7 @@
 #include "common.h"
 #include "../partners.h"
 #include "effects.h"
+#include "functions.h"
```
Comment (ethteck) [inline src/world/partner/lakilester.c:4]: functions.h is part of common.h so this isn't necessary

File: src/world/area_arn/arn_08/events.c [hunk 12]
```diff
@@ -4,7 +4,7 @@
 
 extern Npc* wPartnerNpc;
 void partner_set_goal_pos(s32, s32);
-void func_800EF3D4(s16);
```
Comment (ethteck) [inline src/world/area_arn/arn_08/events.c:7]: this function decl shouldn't be necessary anymore, since it's in functions.h

**Review by ethteck — CHANGES_REQUESTED** [review]
2 smols

**Review by nanaian — APPROVED** [review]

**Review by ethteck — APPROVED** [review]

**Pull request merged**